### PR TITLE
feat: support human-readable output message for findings

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

File: src/findings/finding_container.py [hunk 1]
```diff
@@ -49,6 +49,27 @@ def getAllFindings(cls):
...
+                file_to_findings[finding.location.proto_file_name] = []
+            file_to_findings[finding.location.proto_file_name].append(finding)
+        # Add each finding to the output message.
+        for file_name in file_to_findings:
+            # Customize sort key function to output the findings in the same
+            # file based on the source code line number.
+            def _get_line_number(finding):
+                return finding.location.source_code_line
+
+            file_to_findings[file_name].sort(key=_get_line_number)
```
Comment (software-dov) [inline src/findings/finding_container.py:68]: For a single line key function, it's more compact to just use a lambda.
```python
file_to_findings[file_name].sort(key=lambda f: f.location.source_code_line)
```

File: src/findings/finding_container.py [hunk 2]
```diff
@@ -49,6 +49,27 @@ def getAllFindings(cls):
...
 
+    @classmethod
+    def toHumanReadableMessage(cls):
+        output_message = ""
+        file_to_findings = {}
+        for finding in cls.getActionableFindings():
+            # Create a map to summarize the findings based on proto file name.
+            if finding.location.proto_file_name not in file_to_findings:
+                file_to_findings[finding.location.proto_file_name] = []
+            file_to_findings[finding.location.proto_file_name].append(finding)
```
Comment (software-dov) [inline src/findings/finding_container.py:60]: You can use a `defaultdict` and save a few lines.
```python
file_to_findings = defaultdict(list)
for finding in cls.getActionableFindings():
    file_to_findings[finding.location.proto_file_name].append(finding)
```

File: src/findings/finding_container.py [hunk 3]
```diff
@@ -49,6 +49,27 @@ def getAllFindings(cls):
...
+        # Add each finding to the output message.
+        for file_name in file_to_findings:
```
Comment (software-dov) [inline src/findings/finding_container.py:62]: We can save the lookup in the dict by iterating over `items` instead.
```python
for file_name, findings in file_to_findings.items():
```

**Review by software-dov — COMMENTED** [review]
Looks good, just some style and compactness nits.

File: src/findings/finding_container.py (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply src/findings/finding_container.py]: lamda function in python is new to me, thanks!

File: src/findings/finding_container.py (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply src/findings/finding_container.py]: Sure! Just learned the useful usage for defaultdict from another PR, thanks!

File: src/findings/finding_container.py (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply src/findings/finding_container.py]: Done, much cleaner now. Thanks!

File: src/findings/finding_container.py (hunk 2 above)
Comment (software-dov) [reply src/findings/finding_container.py]: They're my favorite data structure :)

**Review by software-dov — APPROVED** [review]

**Pull request merged**